internal/provider/datasource_subaccount_service_plan_test.go

```diff
@@ -87,24 +89,29 @@ func TestDataSourceSubaccountServicePlan(t *testing.T) {
 
 }
 
-func hclDatasourceSubaccountPlanById(resourceName string, subaccountId string, planId string) string {
+func hclDatasourceSubaccountPlanByIdBySubaccountIdFromFilteredList(resourceName string, subaccountName string, planName string) string {
```

Comment:
This function now selects any service plan with a matching name. For the purpose of the test this should be sufficient I think.

Reply:
I am afraid this won't do the trick. Service plan names are not unique, only in combination with the offering name the plan ID can be clearly identified.